### PR TITLE
new cfDNA sample icon

### PR DESCRIPTION
**Pull request by khzhu**
# What? Why?
To enhance cfDNA sample visualization. Currently, cfDNA samples are labeled with indigo not easily connect to the colours of body fluids.

Changes proposed in this pull request:
- add a new icon for cfDNA sample: a lab test tube filled with coloured liquid. 
- made the colour of liquid will match the type of cfDNA sample. For example, blood plasma; red, pleural effusion; straw, other body fluids; pale yellow. 
- change liquid tumour sample colour from **indigo** to **straw** to reflect the base colour of body fluids

# Checks
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

**Comment by khzhu** [comment]
@inodb , some screenshots.
blood-plasma-cfDNA (1)
other-body-fluid-cfDNA (1)

**Comment by jjgao** [comment]
@khzhu would you please rebase?

**Comment by khzhu** [comment]
@jjgao , will do soon after meeting. thanks!

File: src/appBootstrapper.jsx [hunk 1]
```diff
@@ -50,6 +50,10 @@ initializeConfiguration();
 // THIS TELLS WEBPACK BUNDLE LOADER WHERE TO LOAD SPLIT BUNDLES
 __webpack_public_path__ = AppConfig.frontendUrl;
 
+if (localStorage.heroku && localStorage.localdev !== "true") {
+    __webpack_public_path__ = ['//',localStorage.heroku,'.herokuapp.com','/'].join('');
+}
```
Comment (inodb) [inline src/appBootstrapper.jsx]: this prolly shouldn't be in this PR

File: src/pages/patientView/genomicOverview/Tracks.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ export default class Tracks extends React.Component<TracksPropTypes, {}> {
         // --- end of raphael config ---
 
         // --- chromosome chart ---
-        var chmInfo = tracksHelper.getChmInfo();
+        var chmInfo = tracksHelper.getChmInfo(this.props.mutations[0].ncbiBuild);
```
Comment (inodb) [inline src/pages/patientView/genomicOverview/Tracks.tsx]: this also seems unrelated

File: src/pages/patientView/sampleManager.tsx [hunk 3]
```diff
@@ -181,6 +181,17 @@ class SampleManager {
     getComponentsForSamples() {
         this.samples.map((sample)=>this.getComponentForSample(sample.id));
     }
+
+    getCfDNASampleIconColor(sampleId:string):string {
+        if (["Liquid_Tumor", "Unknown_Tumor"].includes(
+                this.clinicalDataLegacyCleanAndDerived[sampleId]['SAMPLE_TYPE'] )) {
+            return styles.sampleColorMetastasis;
+        } else if (this.clinicalDataLegacyCleanAndDerived[sampleId]['SAMPLE_TYPE'] === "Pleural_Effusion") {
```
Comment (inodb) [inline src/pages/patientView/sampleManager.tsx]: @khzhu I just looked in our database could you change the check to look for pleural effusion in the following other attributes as well:

```
CATEGORY_SAMPLE
PRIMARY_SITE
PROCEDURE_TYPE
SAMPLE_SITE_SOURCE
SITE_OF_RECURRENCE
TISSUE_SITE
TUMOR_TISSUE_SITE
```
Ideally make it a pretty loose case insensitive match for something like `.*pleural.*effusion.*`

File: src/pages/patientView/sampleManager.tsx [hunk 4]
```diff
@@ -181,6 +181,17 @@ class SampleManager {
...
+            return styles.sampleColorCfdna;
```
Comment (inodb) [inline src/pages/patientView/sampleManager.tsx]: this name is a little weird b/c pleural effusion is also a fluid. Maybe better to use e.g. `CfdnaNonBlood` and `CfdnaBlood`? Or `CfdnaPleuralEffusion` and `CfdnaBlood`?

File: src/pages/patientView/genomicOverview/Tracks.tsx (hunk 2 above)
Comment (khzhu) [reply src/pages/patientView/genomicOverview/Tracks.tsx]: will remove.

File: src/appBootstrapper.jsx (hunk 1 above)
Comment (khzhu) [reply src/appBootstrapper.jsx]: will do a rebase and remove those unrelated. sorry!

File: src/pages/patientView/sampleManager.tsx (hunk 3 above)
Comment (khzhu) [reply src/pages/patientView/sampleManager.tsx]: agree! will make changes.

File: src/pages/patientView/sampleManager.tsx (hunk 4 above)
Comment (khzhu) [reply src/pages/patientView/sampleManager.tsx]: sure. will adjust.

**Comment by khzhu** [comment]
thanks, @inodb , will go through the list and address those comments.
